tests/test_optimizer_random_dag.py

```diff
@@ -116,13 +131,13 @@ def compare_optimization_results(dag: sky.Dag, minimize_cost: bool):
                                                       dag.tasks, optimizer_plan)
 
     min_objective = find_min_objective(copy_dag, minimize_cost)
-    assert abs(objective - min_objective) < 1e-3
+    assert abs(objective - min_objective) < 5e-2
```

Comment:
In my original test result, the abs value is ~0.12, bigger than `5e-2`. Do we know why changing to `5e-2` fixes this error?